db_migrator--0.9.0.sql

```diff
@@ -20,25 +20,179 @@ CREATE FUNCTION materialize_foreign_table(
 ) RETURNS boolean
    LANGUAGE plpgsql VOLATILE STRICT SET search_path = pg_catalog AS
 $$DECLARE
-   ft     name;
-   errmsg text;
-   detail text;
+   ft                 name;
+   stmt               text;
+   errmsg             text;
+   detail             text;
+   cur_partitions     refcursor;
+   cur_subpartitions  refcursor;
+   with_partitions    boolean;
+   with_subpartitions boolean;
+   v_schema           text;
+   v_table            text;
+   v_partition        text;
+   v_subpartition     text;
+   v_type             text;
+   v_exp              text;
+   v_pos              int;
+   v_lastpos          int;
+   v_values           text;
+   v_lowervalues      text;
+   v_subvalues        text;
+   v_sublowvalues     text;
+   v_default          boolean;
 BEGIN
    BEGIN
       /* rename the foreign table */
       ft := substr(table_name, 1, 62) || E'\x07';
-      EXECUTE format('ALTER FOREIGN TABLE %I.%I RENAME TO %I', schema, table_name, ft);
+      stmt := format('ALTER FOREIGN TABLE %I.%I RENAME TO %I', schema, table_name, ft);
+      EXECUTE stmt;
+
+      /* create a table or partitioned table */
+      stmt := format('CREATE TABLE %I.%I (LIKE %I.%I)', schema, table_name, schema, ft);
+
+      OPEN cur_partitions FOR EXECUTE format($stmt$
+         SELECT schema, table_name, partition_name,
+               type, expression, position, values, is_default
+            FROM %I.partitions WHERE table_name = $1 AND migrate
+            ORDER BY position $stmt$, pgstage_schema)
+         USING table_name;
+
+      FETCH cur_partitions INTO
+         v_schema, v_table, v_partition,
+         v_type, v_exp, v_pos, v_values, v_default;
+
+      with_partitions := FOUND;
+
+      /* create table as a partitioned table */
+      IF with_partitions THEN
+         IF (v_type = 'HASH') THEN
+            /* retrieve the last partition definition */
+            FETCH LAST FROM cur_partitions INTO
+               v_schema, v_table, v_partition,
+               v_type, v_exp, v_lastpos, v_values, v_default;
+
+            /* go back to the first row */
+            FETCH FIRST FROM cur_partitions INTO
+               v_schema, v_table, v_partition,
+               v_type, v_exp, v_pos, v_values, v_default;
+         END IF;
 
-      /* create a table */
-      EXECUTE format('CREATE TABLE %I.%I (LIKE %I.%I)', schema, table_name, schema, ft);
+         stmt := format('%s PARTITION BY %s (%s)', stmt, v_type, v_exp);
+      END IF;
+      EXECUTE stmt;
+
+      /* create partitions */
+      IF with_partitions THEN
+         LOOP
+            stmt := format(
+               'CREATE TABLE %1$I.%3$I PARTITION OF %1$I.%2$I %4$s',
+               v_schema, v_table, v_partition,
+               CASE
+                  WHEN v_default THEN 'DEFAULT'
+                  WHEN v_type = 'LIST' THEN format(
+                     'FOR VALUES IN (%s)', 
+                     v_values
```

Comment:
There should be a call to `translate_expression_fun` here and in the other branches.
The same applies to subpartitioning.

Reply:
As wrote earlier, I've decided to use translate expression in plugin side because of array construct. Could be enhanced in the future.

Reply:
Finally did it! https://github.com/cybertec-postgresql/db_migrator/pull/11/commits/e0eab88da6e0289233e4bdb2f30794975f740b6f

No regression with ora_migrator, especially expression boundaries defined in the subpartitions test

 https://github.com/cybertec-postgresql/ora_migrator/pull/42/commits/7b3cd03fa4955596da056902e0a9db47e215c161